src/dstore/dstore_base.c

```diff
@@ -31,6 +31,8 @@
 #include "dstore_bufvec.h" /* data buffers and vectors */
 #include "operation.h"
 
+#define DSAL_MAX_IO_SIZE (1024*1024)
```

Comment:
This macro is already defined at line number 125, the reason for defining it was we were not able to do write of more than 1MB but it is not the case with the truncate, you can do a single truncate operation upto 20MB as per old code implementation. Please refer following code snippet from old code.

```
/* NSAL_TUNEABLE:
 * Size of data block that can be definitely UNMAP-ed by Motr
 * without generating errors at RPC layer or getting stuck somewhere
 * in the state machine.
 */
/* Default value: 5120 4K pages or 20 1MB pages or 20MB of data */
static const uint64_t m0_cortxfs_trunc_data_per_request = 20 * (1 << 20);

/** Submits UNMAP requests to Clvois and waits until the data blocks
 * are actually unmapped.
 * NOTE: Motr is not able to handle large extents in the truncate operations,
 * so that we are sending only small portions of extents per request.

```

So you can do two things here.

1. Remove macro define at line number 125 as it is no longer needed and TODO mentioned along with that.
2. Rename this macro specific to truncate operation something like DSAL_MAX_TRUNC_OP_SIZE (20 * 1024 * 1024)  with this you will be submitting lesser truncate request to motr. I don't see any issue with keeping it 20MB as it was already working in old code.

Reply:
Done.